qa/zcash/smoke_tests.py

```diff
@@ -0,0 +1,786 @@
+    ('4ee', True,  True ), # Send funds from t-addr and Sapling z-addr to Sapling z-addr using z_mergetoaddress
+    ('4ff', True,  True ), # Send funds from Sprout z-addr and Sprout z-addr to Sprout z-addr using z_mergetoaddress
+    ('4gg', True,  True ), # Send funds from  Sapling z-addr and Sapling z-addr to Sapling z-addr using z_mergetoaddress
+    # Wallet
+    ('5a', True, True), # After generating multiple z-addrs, run z_listaddresses
+    ('5b', True, True), # Run z_validateaddress with a Sprout z-addr
+    ('5c', True, True), # Run z_validateaddress with a Sapling z-addr
+    ('5d', True, True), # After a transaction, run z_listunspent
+    ('5e', True, True), # After a transaction, run z_listreceivedbyaddress
+    ('5f', True, True), # After a transaction, run z_getbalance
+    ('5g', True, True), # After a transaction, run z_gettotalbalance
+    ('5h', True, True), # Run z_exportkey using a Sprout z-addr
+    ('5i', True, True), # Run z_importkey using the zkey from a Sprout z-addr
+    ('5j', True, True), # Run z_exportkey using a Sapling z-addr
+    ('5k', True, True), # Run z_importkey using the zkey from a Sapling z-addr
+    ('5l', True, True), # Run z_exportwallet
+    ('5m', True, True), # Run z_importwallet
+    ('5n', True, True), # Run z_shieldcoinbase
+    ('5o', True, True), # Run getwalletinfo
+    # Network
+    ('6a', True, True), # Run getpeerinfo
+    ('6b', True, True), # Run getnetworkinfo
+    ('6c', True, False), # Run getdeprecationinfo
+    ('6d', True, True), # Run getconnectioncount
+    ('6e', True, True), # Run getaddednodeinfo
+    # Mining
+    ('7a', True, True), # Run getblocksubsidy
+    ('7b', True, True), # Run getblocktemplate
+    ('7c', True, True), # Run getmininginfo
+    ('7d', True, True), # Run getnetworkhashps
+    ('7e', True, True), # Run getnetworksolps
+]
+
+
+#
+# Test helpers
+#
+
+def run_cmd(results, case, zcash, name, args=[]):
+    print('-----')
+    print('%s $ zcash-cli %s %s' % (
+        case.ljust(3),
+        name,
+        ' '.join([str(arg) for arg in args],
+    )))
+    try:
+        res = zcash.__getattr__(name)(*args)
+        print(res)
+        print()
+        if results is not None and len(case) > 0:
+            results[case] = True
+        return res
+    except JSONRPCException as e:
+        print('error code: %d' % e.error['code'])
+        print('error message:')
+        print(e.error['message'])
+        if results is not None and len(case) > 0:
+            results[case] = False
+        return None
+
+def wait_for_balance(zcash, zaddr, expected=None, timeout=900):
+    print('Waiting for funds to %s...' % zaddr)
+    if expected is not None:
+        print('Expecting %s' % expected)
+    balance = zcash.z_getbalance(zaddr)
+    for _ in iter(range(timeout)):
+        balance = zcash.z_getbalance(zaddr)
+        if expected is None and balance != 0:
+            break
+        if expected is not None and balance == expected:
```

Comment:
If balance != expected, then this will wait until it times out.
The innards of the loop could be simplified to:
```
balance = zcash.z_getbalance(zaddr)
if balance != 0:
    break
```
and it should still work and would prevent this problem.

Reply:
No, this doesn't work for some tests where the target address already has a balance. The reason I added this timeout is because the smoke test necessarily depends on the creation of blocks in the live network, so we can't predict how long it might require.

Reply:
Oops. We do still need the `time.sleep(1)`. I was a bit ambiguous there. What I really meant was replacing
```
balance = zcash.z_getbalance(zaddr)
if expected is None and balance != 0:
    break
if expected is not None and balance == expected:
    break
```
with the above.